syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/datagenerator/DataGeneratorSchemaTransformProvider.java

```diff
@@ -0,0 +1,126 @@
+import com.google.cloud.syndeo.transforms.datagenerator.DataGeneratorSchemaTransformProvider.DataGeneratorSchemaTransformConfiguration;
+import java.util.Collections;
+import java.util.List;
+import org.apache.beam.sdk.schemas.AutoValueSchema;
+import org.apache.beam.sdk.schemas.annotations.DefaultSchema;
+import org.apache.beam.sdk.schemas.annotations.SchemaFieldDescription;
+import org.apache.beam.sdk.schemas.transforms.SchemaTransform;
+import org.apache.beam.sdk.schemas.transforms.SchemaTransformProvider;
+import org.apache.beam.sdk.transforms.SerializableFunction;
+import org.apache.beam.sdk.values.Row;
+import org.apache.commons.lang3.StringUtils;
+import org.checkerframework.checker.initialization.qual.Initialized;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.checker.nullness.qual.UnknownKeyFor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@AutoService(SchemaTransformProvider.class)
+public class DataGeneratorSchemaTransformProvider
+    extends TypedSchemaTransformProvider<DataGeneratorSchemaTransformConfiguration> {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(DataGeneratorSchemaTransformProvider.class);
+
+  @Override
+  public SchemaTransform from(DataGeneratorSchemaTransformConfiguration configuration) {
+    org.apache.beam.sdk.schemas.Schema beamSchema;
+    SerializableFunction<byte[], Row> valueMapper;
+    if (StringUtils.isBlank(configuration.getSchema())) {
+      throw new IllegalArgumentException("No schema specified.");
+    }
+
+    SchemaTransform transform =
+        new DataGeneratorSchemaTransform(
+            configuration.getRecordsPerSecond(),
+            configuration.getMinutesToRun(),
+            configuration.getSchema());
+    return transform;
+  }
+
+  @Override
+  public @UnknownKeyFor @NonNull @Initialized String identifier() {
+    return "syndeo:schematransform:com.google.cloud:data_generator:v1";
+  }
+
+  @Override
+  public @UnknownKeyFor @NonNull @Initialized List<@UnknownKeyFor @NonNull @Initialized String>
+      inputCollectionNames() {
+    return Collections.emptyList();
+  }
+
+  @Override
+  public @UnknownKeyFor @NonNull @Initialized List<@UnknownKeyFor @NonNull @Initialized String>
+      outputCollectionNames() {
+    return List.of("output");
+  }
+
+  @Override
+  public Class<DataGeneratorSchemaTransformConfiguration> configurationClass() {
+    return DataGeneratorSchemaTransformConfiguration.class;
+  }
+
+  @DefaultSchema(AutoValueSchema.class)
+  @AutoValue
+  public abstract static class DataGeneratorSchemaTransformConfiguration {
+
+    @SchemaFieldDescription("The number of records generated per second")
+    public abstract @Nullable Long getRecordsPerSecond();
+
+    @SchemaFieldDescription("The number of minutes to run the generator.")
+    public abstract @Nullable Long getMinutesToRun();
```

Comment:
maybe this should be in seconds rather than minutes? Minutes might be a very large resolution for someone who may want to run a 'short' test of e.g. 30 seconds

Reply:
changed to seconds